src/Memoize.jl

```diff
@@ -12,6 +12,12 @@ macro memoize(args...)
         error("Memoize accepts at most two arguments")
     end
 
+    rettype = :nothin
```

Comment:
I wonder whether it might be better just to set this to `Any`. Then you shouldn't need the branch on line 106 (can just go with the code in `else`).